### ISSUE TITLE
Get covariates for daymet records

### ISSUE THREAD

**Issue by djhocking** [Conte-Ecology/conteStreamTemperature_northeast#3]
I haven't dealt with long format tables before. I need to add the covariates to the daymet records so that each of the parameters below have their own row. Is there an easy way to add this to the query below?

```
var.names <- c("Latitude", "Longitude", "airTemp", "airTempLagged1", "airTempLagged2", "prcp", "prcpLagged1", "prcpLagged2", "prcpLagged3", "dOY", "Forest", "Herbacious", "Agriculture", "Developed", "TotDASqKM", "ReachElevationM", "ImpoundmentsAllSqKM", "HydrologicGroupAB", "SurficialCoarseC", "CONUSWetland", "ReachSlopePCNT", "srad", "dayl", "swe")
```

Query:

```
  qry_daymet <- tbl(db, 'daymet') %>%
    left_join(select(qry_locations, site, featureid), by='featureid') %>%
    filter(featureid %in% catches) %>%
    mutate(airTemp = (tmax + tmin)/2)
```

**Comment by walkerjeffd** [comment]
eh yeah, i'm not exactly sure how to do that with SQL. i think you can, but we might need to get the covariates separately, use tidyr to convert from long to wide, and then merge with the daymet table. i'll give it a shot...

**Comment by djhocking** [comment]
Okay, I'll be doing it in small enough chunks that it shouldn't be a big deal collecting to R then tidying then doing a `left_join`.

**Comment by walkerjeffd** [comment]
something like this? one issue is whether you want local or upstream covariates. does that vary the covariate variable? i just filtered for upstream ones here.

``` r
# covariates
qry_covariates <- select(qry_locations, site, featureid) %>%
  left_join(tbl(db, 'covariates'), by='featureid')

df_covariates_long <- collect(qry_covariates)

df_covariates_wide <- tidyr::spread(df_covariates_long, variable, value)

# which zone? upstream or downstream?
df_covariates_upstream <- filter(df_covariates_wide, zone=="upstream")

masterdata <- left_join(tbl_daymet, df_covariates_upstream, by=c('featureid', 'site'))
```

**Comment by djhocking** [comment]
Houston, we may have a problem. Do we only have these covariates for the catchments with observed data? I'm getting a lot of `NA`. I need these to do predictions to ALL catchments.

**Comment by walkerjeffd** [comment]
uhh, better ask kyle. i haven't really looked at the data, just uploaded what he gave us.

**Comment by djhocking** [comment]
False alarm. I couldn't get a hold of Kyle but I did another check and I think it's okay. At first I just pull the covariates for the first 100 `featureid` and all the covariates except area and elevation were `NA` for almost every catchment. I just did a random sample of 10,000 catchments and there were only 222 with `NA` for `forest` and 473 with `NA` for `surfcoarse`. 

The way  the `featureid` were indexed must have happened to start with ones from an area without GIS data for the catchments.
